src/analysis/special_functions/improper_integrals.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import measure_theory.integral.integral_eq_improper
+import measure_theory.integral.exp_decay
+import analysis.special_functions.integrals
+
+/-!
+# Evaluation of specific improper integrals
+
+This file contains some integrability results, and evaluations of integrals, over `ℝ` or over
+half-infinite intervals in `ℝ`.
+
+## See also
+
+- `analysis.special_functions.integrals` -- integrals over finite intervals
+- `analysis.special_functions.gaussian` -- integral of `exp (-x ^ 2)`
+-/
+
+open real set filter measure_theory
+open_locale topology
+
+lemma integral_exp_neg_Ioi : ∫ (x : ℝ) in Ioi 0, exp (-x) = 1 :=
```

Comment:
Do we have the
```lean
lemma integral_exp_Iio : ∫ (x : ℝ) in Iio 0, exp x = 1 :=
```
variant already?

Reply:
No, I don't think we do. It's not as straightforward as it should be to translate between the two, since we don't have one-shot lemmas for relating `∫ (x : ℝ) in Iio a, f (-x)` and `∫ (x : ℝ) in Ioi (-a), f x`; we have the analogue for finite intervals, and for intervals over all of ℝ, but not for the one-sided integrals. I can make a PR for this if it's wanted.

Reply:
I've added some API for substituting `-x` in integrals over infinite intervals (see #18899, which this is now dependent on) and used it to add the mirror-image lemma. I went for the `Iic` rather than `Iio` version, since that fits our convention that interval integrals are always over `Ioc`; but with the new lemmas at #18899 it is trivial to switch between the two anyway.